coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/functions/HiveGenericUDFReturnTypeInference.java

```diff
@@ -99,7 +99,17 @@ private URLClassLoader getUdfClassLoader() {
   }
 
   private final Class getDynamicallyLoadedUdfClass() throws ClassNotFoundException {
-    return Class.forName(_udfClassName, true, getUdfClassLoader());
+    try {
+      return Class.forName(_udfClassName, true, getUdfClassLoader());
+    } catch (NoClassDefFoundError error) {
+      if (error.getMessage().contains("GenericUDF")) {
+        // If GenericUDF class could not be found, add `hive-exec:core` in `_udfDependencies` to download the missing class
+        _udfClassLoader = null; // set it to null to re-download
+        _udfDependencies.add("com.linkedin.hive:hive-exec:1.1.0.+:core");
```

Comment:
Is there a way to infer this automatically from the UDF dependencies? Upstream dependencies may not match the Hive version, the `com.linkedin.hive` group name, or the `core` classifier.

Reply:
I tried but didn't find a way to infer this automatically from the UDF dependencies, could we change it to `org.apache.hive:hive-exec:1.1.0` so that the upstream dependencies could also match and retrieve it.

Reply:
Yes, if it works internally.

Reply:
Thanks! Updated and it could work internally.